lib/apiservers/engine/backends/container.go

```diff
@@ -761,6 +761,16 @@ func (c *Container) ContainerRm(name string, config *types.ContainerRmConfig) er
 		if state.Status == "Starting" {
 			return derr.NewRequestConflictError(fmt.Errorf("The container is starting.  To remove use -f"))
 		}
+
+		handle, err := c.Handle(id, name)
+		if err != nil {
+			return err
```

Comment:
Is this a docker-typed error?

Reply:
I think so. It returns either `NotFoundError` or `InternalServerError`.